### PR TITLE
Buttons to download credentials

### PR DESCRIPTION
**Pull request by freshteapot**
## Summary

- Able  to download service account credentials, making it easier to setup azure pipeline.
- Able to download container registery, making it easier to setup azure pipeline.

### CODE REVIEW

File: Source/SelfService/Web/documentation/container.tsx [hunk 1]
```diff
@@ -81,6 +82,15 @@ export const DocumentationContainerScreen: React.FunctionComponent<Props> = (pro
                                     Verify access to kubernetes
                                 </a>
                             </li>
+                            <li>
+                                <a href="#" onClick={(event) => {
+                                    event.preventDefault();
+                                    const href = `/documentation/application/${applicationId}/${environment}/ci-cd/-azure-pipelines`;
```
Comment (joelhoisko) [inline Source/SelfService/Web/documentation/container.tsx:88]:
```suggestion
                                    const href = `/documentation/application/${applicationId}/${environment}/ci-cd/azure-pipelines`;
```

Was the `-` prefix intentional? Looks a bit odd to me.

File: Source/SelfService/Web/documentation/container.tsx [hunk 2]
```diff
@@ -107,6 +117,17 @@ export const DocumentationContainerScreen: React.FunctionComponent<Props> = (pro
                         <VerifyKubernetesAccess {...docProps} />
                     </Route>
 
+                    <Route exact path="/documentation/application/:applicationId/:environment/ci-cd/-azure-pipelines">
```
Comment (joelhoisko) [inline Source/SelfService/Web/documentation/container.tsx:120]:
```suggestion
                    <Route exact path="/documentation/application/:applicationId/:environment/ci-cd/azure-pipelines">
```

Same here with the `-` prefix.

File: Source/SelfService/Web/documentation/cicd/setupAzurePipelines.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                const data = await getKubernetesServiceAccount(applicationID);
+                await navigator.clipboard.writeText(data);
+                enqueueSnackbar('Kubernetes service account copied to clipboard.');
+            } catch {
+                enqueueSnackbar('Failed to get data.', { variant: 'error' });
+            }
+        }}
+    >
+        Copy Credentials to clipboard
+    </DownloadButton>;
```
Comment (joelhoisko) [inline Source/SelfService/Web/documentation/cicd/setupAzurePipelines.tsx:30]: The `<DownloadButton>`'s look a bit weird as you're just copying credentials, but I also know that mui v4 icons don't have the nice copy icon yet. In the future we should make a `<CopyButton>` for this. Just pointing this out as a paper trail https://app.asana.com/0/1200181647276434/1201312046888451

**Review by joelhoisko — CHANGES_REQUESTED** [review]
Thinking about this, the information here is not specific to Azure Pipelines at all so maybe we could just call it something like "CI/CD Credentials" or something.

On the other hand, content is king and when we get something a bit nicer we can do some renaming.

Otherwise LGTM, just a small point about the URL.

File: Source/SelfService/Web/api/cicd.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        method: 'GET',
+        mode: 'cors',
+    });
+
+    if (response.status !== 200) {
+        console.error(response);
+        throw Error('Failed to service account');
+    }
+
+    return await response.json() as any;
```
[the remark this answers is not in the document]
Comment (freshteapot) [reply Source/SelfService/Web/api/cicd.ts]: I have changed this to use a type not any, for both (you see it in my latest commit)

At least on my vscode, my opting to await is throwing errors when I remove the casting.
So I have kept it.

File: Source/SelfService/Web/documentation/container.tsx (hunk 1 above)
Comment (freshteapot) [reply Source/SelfService/Web/documentation/container.tsx]: Fixed, good spot.

File: Source/SelfService/Web/documentation/cicd/setupAzurePipelines.tsx (hunk 3 above)
Comment (freshteapot) [reply Source/SelfService/Web/documentation/cicd/setupAzurePipelines.tsx]: Changed to ButtonText.

File: Source/SelfService/Web/documentation/container.tsx (hunk 2 above)
Comment (freshteapot) [reply Source/SelfService/Web/documentation/container.tsx]: Fixed, good spot.

File: Source/SelfService/Web/api/cicd.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
```
[the remark this answers is not in the document]
Comment (freshteapot) [reply Source/SelfService/Web/api/cicd.ts]: I have changed this to use a type not any, for both (you see it in my latest commit)

At least on my vscode, my opting to await is throwing errors when I remove the casting.
So I have kept it.

**Review by joelhoisko — APPROVED** [review]
👍

**Comment by joelhoisko** [comment]
Also I already marked our conversation about the `as any` stuff as resolved, but here's a nice example explaining the some of the difference between the return type annotation vs using the `as type` stuff. https://stackoverflow.com/a/61386586/5806412

**Pull request merged**